src/DynamoCoreWpf/ViewModels/Watch3D/AttachedProperties.cs

```diff
@@ -278,6 +377,42 @@ public static bool IsSpecialRenderPackage(DependencyObject element)
 
         //#endregion
 
+        #region utils
+        
+        /// <summary>
+        /// Sets all colors on Geometry to a single color.
+        /// </summary>
+        /// <param name="geom"></param>
+        /// <param name="color"></param>
+        private static void SetAllColors(GeometryModel3D geom, Color4 color)
+        {
+            var newColorCollection = new Color4Collection(Enumerable.Repeat(color, geom.Geometry.Colors.Count));
+            geom.Geometry.Colors = newColorCollection;
```

Comment:
So we still have to figure out why this doesn't work for meshes?

Reply:
yes, I have just grabbed that task, I think it has to do with the shader we are using.